### DISCUSSION TITLE
Procedural lights

### DISCUSSION THREAD

**Discussion by apofex** [godotengine/godot-proposals#6842] [2D]
I think Godot must have procedural 2d lights like in Unity. Now to make the point light in Godot, you must make the texture. It's not always convenient during level creating.

**Comment by AThousandShips** [comment]
Can you please elaborate at least some on what this means? I know this isn't a proposal where a lot of detail has to be given but this isn't really anything to go on, please don't expect people to go and look up what you mean

**Reply by apofex** [reply]
Now to make the point light in Godot, you must make the texture. It's not always convenient during level creating.

**Reply by Zireael07** [reply]
Godot can procedurally create textures

**Reply by AThousandShips** [reply]
Just use a radial gradient texture?

**Reply by AThousandShips** [reply]
As mentioned in the [documentation](https://docs.godotengine.org/en/stable/tutorials/2d/2d_lights_and_shadows.html#point-lights)

**Reply by apofex** [reply]
Sorry, I didn't see it. But in Unity we have possibilities to edit light directly in viewport.

**Reply by Zireael07** [reply]
Godot's philosophy is to be lean. It can't do everything Unity does nor in the same way, it would unavoidably bloat the engine
